dbus/dbus.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package dbus
+
+import "github.com/pkg/errors"
+
+var dbusAPI DBusAPI = nil
+
+type DBusAPI interface {
+	GenerateGUID() string
+	IsGUID(string) bool
+}
+
+func GetDBusAPI() (DBusAPI, error) {
+	if dbusAPI != nil {
+		return dbusAPI, nil
+	}
+	return nil, errors.New("no D-Bus interface available")
```

Comment:
Btw, this is not important at all, but you could define this function twice, once with `nodbus` and once with `!nodbus`. Then you could return the answer directly and you don't need the global.

Reply:
haha, where have I seen this idea before

Reply:
Yes, that is also a valid approach. I wanted to avoid an extra file with a single function.

If during implementation turns out that there are more methods that need to be available for both cases, then we it would make sense to rework as you said.

Reply:
> haha, where have I seen this idea before

Not sure if I understood this, it's not the same approach as you suggested. Though you could say it's inspired by it. This would still only include one function call which returns error, the API getter, and all the other functions calls would only have real implementations, no nil implementations.

But yes, it's fine the way it is as well.

Reply:
> Not sure if I understood this, it's not the same approach as you suggested. Though you could say it's inspired by it. 

That is what I meant :)